### PR TITLE
Asadchev/refactor/lapack

### PR DESCRIPTION
**Pull request by asadchev**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -11,8 +11,9 @@ variables:
   OMPI_MCA_btl_vader_single_copy_mechanism: none
 
 before_script:
-  #- apt-get install -yq ...
   - echo 'localhost slots=8' > /etc/openmpi/openmpi-default-hostfile 
+  - apt-get install -yq intel-mkl-core-c-2020.4-304
+  - ln -s /opt/intel/compilers_and_libraries_2020.4.304/linux/mkl/ /opt/intel/
```
Comment (evaleev) [inline .gitlab-ci.yml:16]: better to load `/opt/intel/compilers_and_libraries_2020.4.304/linux/mkl/bin/mklvars.sh` to test discovery from `MKLROOT`-defined location?

File: src/TiledArray/math/linalg/rank-local.cpp [hunk 2]
```diff
@@ -23,26 +23,33 @@
...
   if constexpr (std::is_same_v<numeric_type, double>)                          \
-    TA_LAPACKE_CALL(d##name, args);                                            \
+                 TA_LAPACK_CALL(d##name, args);                                \
   else if constexpr (std::is_same_v<numeric_type, float>)                      \
-    TA_LAPACKE_CALL(s##name, args);                                            \
-  else if constexpr (std::is_same_v<numeric_type, std::complex<double>>)       \
-    TA_LAPACKE_CALL(z##name, args);                                            \
-  else if constexpr (std::is_same_v<numeric_type, std::complex<float>>)        \
-    TA_LAPACKE_CALL(c##name, args);                                            \
+                 TA_LAPACK_CALL(s##name, args);                                \
```
Comment (evaleev) [inline src/TiledArray/math/linalg/rank-local.cpp:52]: why no complex?

File: .gitlab-ci.yml (hunk 1 above)
Comment (asadchev) [reply .gitlab-ci.yml:16]: i had that at first, but this seems more common case

File: src/TiledArray/math/linalg/rank-local.cpp (hunk 2 above)
Comment (asadchev) [reply src/TiledArray/math/linalg/rank-local.cpp:52]: compile issues due to constexpr defificencies.

**Review by evaleev — APPROVED** [review]

**Pull request merged**